modules/collection/programs/fish.nix

```diff
@@ -124,6 +124,26 @@ in {
     plugins = mkOption {
       type = attrsOf path;
       default = {};
+      description = ''
+        An attrset of plugins.
+
+        In the case where a plugin is a 'vendored' plugin, all we are doing is adding that
+        plugin to `hjem.users.<name>.packages`. As fish automatically discovers those files.
+
+        A vendored plugin is denoted by the existence of one of the following files in its derivation:
+        - /share/fish/vendor_conf.d
+        - /share/fish/vendor_completions.d
+        - /share/fish/vendor_functions.d
+
+        This will be the case with plugins present in `pkgs.fishPlugins`.
+
+        For the remaining cases, a file will be created at `~/.config/fish/conf.d/rum-plugin-<name>.fish`.
+        It will attempt to handle or source a variety of expected files from the derivation. Those files are:
```

Comment:
What happens if the plugin is not standard and does not get handled correctly?

Reply:
Nothing, as in, the files will just not get sourced/the env variable not updated.

Reply:
Do you think adding a sentence explaining this is necessary? As we do not make it error out

Reply:
Added a little something
> If a plugin seems to not work, verify that it contains one of the aformentioned files.